user_docs/markdownTranslate.py

```diff
@@ -0,0 +1,670 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2024 NV Access Limited.
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+from typing import Generator
+import tempfile
+import os
+import contextlib
+import lxml.etree
+import argparse
+import uuid
+import re
+from itertools import zip_longest
+from xml.sax.saxutils import escape as xmlEscape
+from xml.sax.saxutils import unescape as xmlUnescape
+import difflib
+from dataclasses import dataclass
+import subprocess
+
+RAW_GITHUB_REPO_URL = "https://raw.githubusercontent.com/nvaccess/nvda"
+re_kcTitle = re.compile(r"^(<!--\s+KC:title:\s*)(.+?)(\s*-->)$")
+re_kcSettingsSection = re.compile(r"^(<!--\s+KC:settingsSection:\s*)(.+?)(\s*-->)$")
+re_comment = re.compile(r"^<!--.+-->$")
+re_heading = re.compile(r"^(#+\s+)(.+?)((?:\s+\{#.+\})?)$")
+re_bullet = re.compile(r"^(\s*\*\s+)(.+)$")
+re_number = re.compile(r"^(\s*[0-9]+\.\s+)(.+)$")
+re_hiddenHeaderRow = re.compile(r"^\|\s*\.\s*\{\.hideHeaderRow\}\s*(\|\s*\.\s*)*\|$")
+re_postTableHeaderLine = re.compile(r"^(\|\s*-+\s*)+\|$")
+re_tableRow = re.compile(r"^(\|)(.+)(\|)$")
+re_translationID = re.compile(r"^(.*)\$\(ID:([0-9a-f-]+)\)(.*)$")
```

Comment:
I think these should be grouped as an enum

```py
class Re(re.Pattern, Enum):
   KC_TITLE  = re.compile(r"^(<!--\s+KC:title:\s*)(.+?)(\s*-->)$")

Reply:
@michaelDCurran this is the only remaining feedback from me

Reply:
A few issues when trying to do this:
* At runtime:
```
class MarkdownRegex(re.Pattern, enum.Enum):                                                                                                             
  File "C:\Users\mick\AppData\Local\Programs\Python\Python311-32\Lib\enum.py", line 562, in __new__                                                         
    enum_class = super().__new__(metacls, cls, bases, classdict, **kwds)                                                                                    
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                    
TypeError: type 're.Pattern' is not an acceptable base type 
```
* In VS code, Pylance says:
```
Error: Base class "Pattern[Unknown]" is marked final and cannot be subclassed at line 25 and character 21. generated by Pylance
```
* If I don't inherit from re.Pattern however, then I of course get errors in VS Code like:
```
Error: Cannot access attribute "match" for class "Literal[MarkdownRegex.HIDDEN_HEADER_ROW]"   Attribute "match" is unknown at line 93 and character 38. generated by Pylance
```
So based on this, for now perhaps we should leave them as stand-alone constants?

Reply:
Makes sens